source/azure_iot.c

```diff
@@ -0,0 +1,410 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file azure_iot.c
+ * @brief -------.
+ *
+ */
+
+#include "azure_iot.h"
+
+static const char _cAzureIoTBase64Array[] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+/*-----------------------------------------------------------*/
+
+static void prvAzureIoTLogListener( az_log_classification xClassification,
+                                    az_span xMessage )
+{
+    ( void ) xClassification;
+    //In case logs are stripped out, suppress unused parameter error.
+    ( void ) xMessage;
+
+    AZLogInfo( ( "%.*s", az_span_size( xMessage ), az_span_ptr( xMessage ) ) );
+}
+/*-----------------------------------------------------------*/
+
+static AzureIoTResult_t prvAzureIoTBase64Decode( char * pcBase64name,
+                                                uint32_t ulLength,
+                                                uint8_t * pucName,
+                                                uint32_t ulNameSize,
+                                                uint32_t * pulBytesCopied )
+{
+    uint32_t i, j;
```

Comment:
lets name these variable too with the coding style.

Reply:
ehhh do we have to here? I understand it's not consistent but I feel like this is a pretty common practice. Otherwise its `ulI` and `ulJ` or something like that.

Reply:
resolving with not changing unless someone is really opposed